File: test2.py
numBit = 0
sum = 0
sumBit = 0

num = input()

for i in range(len(num)):
  numBit = eval(num[i])
  sum = sum + numBit

sumS = str(sum)

for i in range(len(sumS), 0, -1):
  sumBit = sum // pow(10, i - 1)
  if(sumBit == 0):
    print("ling", end = "")
  elif(sumBit == 1):
    print("yi", end = "")
  elif(sumBit == 2):
    print("er", end = "")
  elif(sumBit == 3):
    print("san", end = "")
  elif(sumBit == 4):
    print("si", end = "")
  elif(sumBit == 5):
    print("wu", end = "")
  elif(sumBit == 6):
    print("liu", end = "")
  elif(sumBit == 7):
    print("qi", end = "")
  elif(sumBit == 8):
    print("ba", end = "")
  elif(sumBit == 9):
    print("jiu", end = "")

  sum = sum - sumBit * pow(10, i - 1)
  if(i > 1):
    print(" ", end = "")

File: test2.cpp
#include<iostream>
#include<string>
#include<math.h>

int main()
{
  int numBit = 0, sum = 0, sumBit = 0;
  std::string num, sumS;

  std::cin >> num;

  for(int i = 0; i < num.size(); i++)
  {
    numBit = (int)num[i] - 48;
    sum = sum + numBit;
  }

  sumS = std::to_string(sum);

  for(double i = sumS.size(); i > 0; i--)
  {
    sumBit = sum / pow(10, i - 1);
    switch(sumBit)
    {
      case 0:
        std::cout << "ling";
        break;
      case 1:
        std::cout << "yi";
        break;
      case 2:
        std::cout << "er";
        break;
      case 3:
        std::cout << "san";
        break;
      case 4:
        std::cout << "si";
        break;
      case 5:
        std::cout << "wu";
        break;
      case 6:
        std::cout << "liu";
        break;
      case 7:
        std::cout << "qi";
        break;
      case 8:
        std::cout << "ba";
        break;
      case 9:
        std::cout << "jiu";
        break;
    }
    sum = sum - sumBit * pow(10, i - 1);
    if(i > 1)
      std::cout << " ";
  }

  std::cout << std::endl;
  system("pause");
}
